File: tests/integration/construct_component_id_output/python/component.py
# Copyright 2016-2024, KhulnaSoft Ltd.  All rights reserved.

from typing import Any, Optional

import codeinfra

class Component(codeinfra.ComponentResource):
    def __init__(self, name: str, id: codeinfra.Input[str], opts: Optional[codeinfra.ResourceOptions] = None):
        props = dict()
        props["id"] = id
        super().__init__("testcomponent:index:Component", name, props, opts, True)

    @property
    @codeinfra.getter
    def id(self) -> codeinfra.Output[str]:
        return codeinfra.get(self, "id")

File: tests/integration/construct_component_plain/nodejs/component.ts
// Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

export interface ComponentArgs {
    children?: number;
}

export class Component extends codeinfra.ComponentResource {
    constructor(name: string, args: ComponentArgs, opts?: codeinfra.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, args, opts, true);
    }
}
